appservice/src/intTest/groovy/nu/yona/server/BasicBuddyTest.groovy

```diff
@@ -1,536 +1,378 @@
+			getAnonMessagesRichardResponse.responseData._embedded.goalConflictMessages[0].url == null
 
-	def 'Richard removes Bob as buddy'() {
-		given:
-		when:
-		def response = appService.removeBuddy(richardQuinBobBuddyURL, richardQuin.password, "Bob, as you know our ways parted so I'll remove you as a buddy.")
+			def getAnonMessagesBobResponse = appService.getAnonymousMessages(bob)
+			getAnonMessagesBobResponse.status == 200
+			getAnonMessagesBobResponse.responseData._embedded.goalConflictMessages.size() == 1
+			getAnonMessagesBobResponse.responseData._embedded.goalConflictMessages[0].nickname == "<self>"
+			getAnonMessagesBobResponse.responseData._embedded.goalConflictMessages[0].goalName == "gambling"
+			getAnonMessagesBobResponse.responseData._embedded.goalConflictMessages[0].url == "http://www.poker.com"
 
-		then:
-		response.status == 200
+		cleanup:
+			appService.deleteUser(richard)
+			appService.deleteUser(bob)
 	}
 
-	def 'Bob checks his direct messages and will find a remove buddy message'(){
+	def 'Richard removes Bob as buddy, so goal conflicts from Bob are gone'()
+	{
 		given:
+			def richardAndBob = addRichardAndBobAsBuddies()
+			def richard = richardAndBob.richard
+			def bob = richardAndBob.bob
+			analysisService.postToAnalysisEngine(richard, ["news/media"], "http://www.refdag.nl")
+			analysisService.postToAnalysisEngine(bob, ["Gambling"], "http://www.poker.com")
+			def buddy = appService.getBuddies(richard)[0]
 
 		when:
-		def response = appService.getAnonymousMessages(bobDunn.url, bobDunn.password)
-		if (response.responseData._embedded && response.responseData._embedded.buddyDisconnectMessages) {
-			bobDunnBuddyRemoveMessageProcessURL = response.responseData._embedded.buddyDisconnectMessages[0]._links.process.href
-		}
+			def response = appService.removeBuddy(richard, buddy, "Bob, as you know our ways parted, so I'll remove you as buddy.")
 
 		then:
-		response.status == 200
-		response.responseData._embedded.buddyDisconnectMessages[0].reason == "USER_REMOVED_BUDDY"
-		response.responseData._embedded.buddyDisconnectMessages[0].nickname == "${richardQuin.nickname}"
-		response.responseData._embedded.buddyDisconnectMessages[0].message == "Bob, as you know our ways parted so I'll remove you as a buddy."
-		response.responseData._embedded.buddyDisconnectMessages[0]._links.self.href.startsWith(bobDunn.url + appService.ANONYMOUS_MESSAGES_PATH_FRAGMENT)
-		bobDunnBuddyRemoveMessageProcessURL.startsWith(response.responseData._embedded.buddyDisconnectMessages[0]._links.self.href)
-	}
+			response.status == 200
+			appService.getBuddies(richard).size() == 0 // Buddy removed for Richard`
+			appService.getBuddies(bob).size() == 1 // Buddy not yet removed for Bob (not processed yet)
 
-	def 'Test what happens if the classification engine detects a potential conflict for Bob (third conflict message) when the buddy disconnect is not yet processed'(){
```

Comment:
The test ``After Richard removed Bob as buddy, new goal conflicts are not reported to the buddies anymore`` does that.